File: app/src/main/java/com/cn29/aac/ui/common/PermissionComponentBuilder.java
package com.cn29.aac.ui.common;

import androidx.fragment.app.Fragment;

public class PermissionComponentBuilder {

  private final Fragment fragment;
  private String[] permissions;
  private int requestCode;

  public PermissionComponentBuilder(Fragment fragment) {
    this.fragment = fragment;
  }


  public PermissionComponentBuilder setPermissions(String[] permissions) {
    this.permissions = permissions;
    return this;
  }

  public PermissionComponentBuilder setRequestCode(int requestCode) {
    this.requestCode = requestCode;
    return this;
  }

  public FragmentPermissionComponent createPermissionComponent() {
    return new FragmentPermissionComponent(fragment, permissions, requestCode);
  }
}

File: app/src/main/java/com/cn29/aac/ui/common/PermissionComponentBuilder.kt
package com.cn29.aac.ui.common

import androidx.fragment.app.Fragment

class PermissionComponentBuilder(private val fragment: Fragment) {
    private lateinit var permissions: Array<String>
    private var requestCode = 0
    fun setPermissions(permissions: Array<String>): PermissionComponentBuilder {
        this.permissions = permissions
        return this
    }

    fun setRequestCode(requestCode: Int): PermissionComponentBuilder {
        this.requestCode = requestCode
        return this
    }

    fun createPermissionComponent(): FragmentPermissionComponent {
        return FragmentPermissionComponent(fragment, permissions, requestCode)
    }

}